projects/packages/publicize/src/social-image-generator/class-setup.php

```diff
@@ -73,22 +73,22 @@ public function generate_token_on_save( $post_id, $post, $update ) {
 			return;
 		}
 
-		if ( ! $publicize->has_social_image_generator_feature() ) {
-			return;
-		}
+		$settings = new Settings();
 
-		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post_id ) ) {
+		if ( ! $settings->is_available() ) {
 			return;
 		}
 
-		if ( ! ( new Settings() )->is_enabled() ) {
+		if ( wp_is_post_autosave( $post ) || wp_is_post_revision( $post_id ) ) {
 			return;
 		}
 
+		// Set SIG to be enabled by default for new posts if the toggle is on.
 		$post_settings = new Post_Settings( $post_id );
 		if (
 			! $update &&
 			'auto-draft' === $post->post_status &&
+			$settings->is_enabled() &&
```

Comment:
This line ensures that if the SIG toggle is enabled on the admin page, SIG gets enabled by default for new posts.